### PR TITLE
Bug 1131442 - Scan GAIA_APPDIRS with js

### PR DESCRIPTION
**Pull request by cctuan**
https://bugzilla.mozilla.org/show_bug.cgi?id=1131442

### CODE REVIEW

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=25ddee521fff)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=2cfdac011f41)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=34be4b844ca3)

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4028ad3db402)

File: build/scan-appdir.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+    return false;
+  } catch (e) {
+    return false;
+  }
+};
+
+ScanAppdir.prototype.execute = function() {
+  var baseSrc;
+  this.appList.split('\n').forEach(function(src) {
+
```
Comment (RickyChien) [inline build/scan-appdir.js]: remove this line

File: build/scan-appdir.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    } else {
+      this.scanAppAndDistributionFolder(src, false);
+    }
+  }, this);
+};
+
+function execute() {
+  var scanAppDir = new ScanAppdir();
+  scanAppDir.execute();
+  dump(scanAppDir.allAppList.join(' '));
```
Comment (RickyChien) [inline build/scan-appdir.js]: We should use utils.log

File: build/scan-appdir.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    }, this);
+    utils.ls(gaiaAppFolder, false).forEach(function(subFolder) {
+      this.pushPathSrc(subFolder);
+    }, this);
+    try {
+      srcAppFolder = utils.getFile(src);
+      utils.ls(srcAppFolder, false).forEach(function(subFolder) {
+        this.pushPathSrc(subFolder);
+      }, this);
+    } catch (e) {}
```
Comment (RickyChien) [inline build/scan-appdir.js]: Add a comment to remind that we should throw an error after Makefile is replaced with JS module.

File: build/scan-appdir.js (hunk 2 above)
Comment (cctuan) [inline build/scan-appdir.js]: utils.log will generate log as '[//LOG]' , we should use dump here.

**Comment by try-server-hook** [comment]
[cctuan George (cctuan)](https://github.com/cctuan) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=24becc93a89d)

**Pull request merged**